File: src/main/java/org/noxet/noxetserver/playerstate/properties/PSPHealthScaled.java
package org.noxet.noxetserver.playerstate.properties;

import org.bukkit.entity.Player;
import org.noxet.noxetserver.playerstate.PlayerStateProperty;

public class PSPHealthScaled implements PlayerStateProperty<Boolean> {
    @Override
    public String getConfigName() {
        return "health_scaled";
    }

    @Override
    public Boolean getDefaultSerializedProperty() {
        return false;
    }

    @Override
    public Boolean getSerializedPropertyFromPlayer(Player player) {
        return player.isHealthScaled();
    }

    @Override
    public void restoreProperty(Player player, Boolean scaled) {
        player.setHealthScaled(scaled);
    }
}

File: src/main/java/org/noxet/noxetserver/playerstate/properties/PSPHealthScaled.scala
package org.noxet.noxetserver.playerstate.properties

import org.bukkit.entity.Player
import org.noxet.noxetserver.playerstate.PlayerStateProperty

object PSPHealthScaled extends PlayerStateProperty[Boolean]:
  override def getConfigName: String = "health_scaled"

  override def getDefaultSerializedProperty: Boolean = false

  override def getSerializedPropertyFromPlayer(player: Player): Boolean =
    player.isHealthScaled

  override def restoreProperty(player: Player, scaled: Boolean): Unit =
    player.setHealthScaled(scaled)